recipe-server/normandy/recipes/decorators.py

```diff
@@ -1,14 +1,15 @@
-class CurrentRevisionProperty(object):
-    """
-    A class to help create class properties that return None if an instance of the parent class
-    has no `current_revision`.
-    """
-    def __init__(self, method):
+class GenericRevisionProperty(object):
+    related_field = None
```

Comment:
I think if related_field was passed as a parameter to __init__, then you could get rid of all the subclasses, and just create instances of this field in the decorators. It would be less lines of code, but I'm not sure if there is something I'm missing that it would break.